lib/omise/OmiseSearch.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+require_once dirname(__FILE__).'/res/OmiseApiResource.php';
+
+/**
+ * This class is not intended to be used directly from client code.
+ *
+ * @see OmiseCharge::search()
+ */
+class OmiseSearch extends OmiseApiResource
+{
+    const ENDPOINT = 'search';
+
+    private $dirty = true;
```

Comment:
I meant `true`. The object is intended to be initially dirty to make sure that the first access to array element will trigger `retrieve()`. However, the encapsulation `protected function __constructor(...)` will override this default value. So it does not matter whether this value is `true` or `false`.

```php
protected function __construct($scope, $publickey = null, $secretkey = null)
{
    parent::__construct($publickey, $secretkey);
    $this->dirty = true;
    $this->scope = $scope;
}
```